machine/v1beta1/types_gcpprovider.go

```diff
@@ -143,10 +147,14 @@ type GCPMachineProviderSpec struct {
 	// +optional
 	ShieldedInstanceConfig GCPShieldedInstanceConfig `json:"shieldedInstanceConfig,omitempty"`
 
-	// confidentialCompute Defines whether the instance should have confidential compute enabled.
-	// If enabled OnHostMaintenance is required to be set to "Terminate".
+	// ConfidentialCompute Defines whether the instance should have confidential compute enabled or not, and the confidential computing technology of choice.
+	// If Disabled, the machine will not be configured to be a confidential computing instance.
+	// If Enabled, confidential computing will be configured and AMD Secure Encrypted Virtualization will be configured by default. That is subject to change over time. If using AMD Secure Encrypted Virtualization is vital, use AMDEncryptedVirtualization explicitly instead.
+	// If AMDEncryptedVirtualization, it will configure AMD Secure Encrypted Virtualization (AMD SEV) as the confidential computing technology.
+	// If AMDEncryptedVirtualizationNestedPaging, it will configure AMD Secure Encrypted Virtualization Secure Nested Paging (AMD SEV-SNP) as the confidential computing technology.
+	// If enabled (any value other than Disabled) OnHostMaintenance is required to be set to "Terminate".
```

Comment:
Is there any validation in place that enforces that `OnHostMaintenance` is set to `Terminate`?

Also, references to fields should use the serialized JSON field format: `OnHostMaintenance` --> `onHostMaintenance`

Reply:
It doesn't hurt to document this as well. It looks like some of it depends on the CPU platform that is chosen. Is there currently a way for users to specify the CPU platform? Is there some validation in place to ensure you aren't attempting to set an invalid confidential computing technology on a CPU platform it wouldn't be supported on?

Would you mind also sharing links to the webhook validations that are in place?